app/models/answer.rb

```diff
@@ -1,3 +1,17 @@
 class Answer < ApplicationRecord
+  ANSWER_LIMIT = 4
   belongs_to :question
+  validates :title, presence: true
+  validate :question_answers
+
+  scope :correct, -> { where(correct: true) }
+
+  private
+
+  def question_answers
+    answers_count = Answer.where(question_id: question_id).count
+    if answers_count == ANSWER_LIMIT
```

Comment:
```suggestion
    if answers_count >= ANSWER_LIMIT
```
я бы из паранойи использовал `>=` 
например такой сценарий:
- в базе 3 записи и два пользователя одновременно создают новый ответ с разных компов
- на момент создания - это валидная операция для каждого из них
- в базе в итоге 5 ответов и это не очень хорошо
-  что еще хуже, это то, что можно продолжить создавать ответы, потому что ответов не 4, а в проверке строгое равенство